pkg/minikube/command/kic_runner.go

```diff
@@ -209,6 +205,24 @@ func (k *kicRunner) Copy(f assets.CopyableFile) error {
 	return k.copy(tf.Name(), dst)
 }
 
+// tempDirectory returns the directory to use as the temp directory
+// or an empty string if it should use the os default temp directory.
+func tempDirectory(isSnap bool) (string, error) {
+	if !isSnap {
```

Comment:
how about instead of passing the the bool, we just call snap detection inside the func ?
also lets choose a better name
how about ?
```
func tmpDirPath () (string, error)
 ```

Reply:
The problem with that is it makes it very hard to test the snap path, which is really the whole point of this PR.